### PR TITLE
feat(ui5-menu): provide busy indicator display

### PR DESCRIPTION
**Pull request by unazko**
New properties "busy" and "busyDelay" have been introduced.
Through those properties the application developers
will be able to configure a busy indicator display for a particular ui5-menu
DOM representation. That way there will be a busy indicator while
the corresponding menu item texts are getting fetched from a database.

Fixes: #6131

### CODE REVIEW

**Review by ilhan007 — CHANGES_REQUESTED** [review]
Clicking on "Request changes" just to be sure the core team have some time to do API review of this feature, @tsanislavgatev @NHristov-sap please continue with the review normally

File: packages/main/test/specs/Menu.spec.js [hunk 1]
```diff
@@ -131,6 +131,25 @@ describe("Menu interaction", () => {
...
+		menuItem.click();
+
+		const subMenuPopover = await browser.$("ui5-static-area-item:last-of-type").shadow$("ui5-responsive-popover");
+		const busyIndicator = await subMenuPopover.shadow$("ui5-busy-indicator");
+
+		assert.strictEqual(await busyIndicator.getAttribute("active"), "true", "Active attribute is propelry set.");
+		assert.strictEqual(await busyIndicator.getAttribute("text"), "Loading...", "Text attribute is propelry set.");
+		assert.strictEqual(await busyIndicator.getAttribute("title"), "Loading...", "Title attribute is propelry set.");
+		assert.strictEqual(await busyIndicator.getAttribute("size"), "Medium", "Size attribute is propelry set.");
+		assert.strictEqual(await busyIndicator.getAttribute("delay"), "100", "Delay attribute is propelry set.");
```
Comment (DavidMitreski5) [inline packages/main/test/specs/Menu.spec.js:151]: "properly"

**Review by DavidMitreski5 — APPROVED** [review]

File: packages/main/test/specs/Menu.spec.js [hunk 2]
```diff
@@ -131,6 +131,23 @@ describe("Menu interaction", () => {
...
+		openButton.click();
+		const menuPopover = await browser.$("ui5-static-area-item:last-of-type").shadow$("ui5-responsive-popover");
+		const visualMenuItem = await menuPopover.$("ui5-li[accessible-name='Open']");
+
+		visualMenuItem.click();
+		const subMenuPopover = await browser.$("ui5-static-area-item:last-of-type").shadow$("ui5-responsive-popover");
+		const busyIndicator = await subMenuPopover.$("ui5-busy-indicator");
+
+		assert.ok(await busyIndicator.getAttribute("active"), "Active attribute is properly set.");
+		assert.strictEqual(await busyIndicator.getAttribute("size"), "Medium", "Size attribute is properly set.");
```
Comment (nnaydenow) [inline packages/main/test/specs/Menu.spec.js:148]: getAttribute => getProperty

**Review by nnaydenow — CHANGES_REQUESTED** [review]

File: packages/main/src/Menu.ts [hunk 3]
```diff
@@ -180,6 +183,30 @@ class Menu extends UI5Element {
 	@property({ type: Boolean })
 	open!:boolean;
 
+	/**
+	 * Defines if a loading indicator would be displayed inside the corresponding ui5-menu popover.
+	 *
+	 * @name sap.ui.webc.main.List.prototype.busy
```
[the remark this answers is not in the document]
Comment (unazko) [reply packages/main/src/Menu.ts:189]: done

File: packages/main/src/MenuItem.ts [hunk 4]
```diff
@@ -96,6 +97,32 @@ class MenuItem extends UI5Element {
...
+	/**
+	 * Defines if a loading indicator would be displayed inside the corresponding sub-menu popover.
+	 *
+	 * Note: If set to <code>true</code> a <code>ui5-busy-indicator</code> component will be displayed into the related to the current <code>ui5-menu-item</code> sub-menu popover.
+	 *
+	 * @name sap.ui.webc.main.List.prototype.busy
+	 * @type {boolean}
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.14.0
```
[the remark this answers is not in the document]
Comment (unazko) [reply packages/main/src/MenuItem.ts:109]: done

File: packages/main/src/Menu.ts [hunk 5]
```diff
@@ -180,6 +183,30 @@ class Menu extends UI5Element {
...
+	 * @type {boolean}
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.13.0
+	 */
+	@property({ type: Boolean })
+	busy!: boolean;
+
+	/**
+	 * Defines the delay in milliseconds, after which the busy indicator will displayed inside the corresponding ui5-menu popover.
```
Comment (DavidMitreski5) [inline packages/main/src/Menu.ts:199]: Defines the delay in milliseconds, after which the busy indicator will be displayed inside the corresponding ui5-menu popover.

File: packages/main/src/MenuItem.ts [hunk 6]
```diff
@@ -96,6 +97,32 @@ class MenuItem extends UI5Element {
 	@property({ type: Boolean })
 	disabled!: boolean;
 
+	/**
+	 * Defines if a loading indicator would be displayed inside the corresponding sub-menu popover.
+	 *
+	 * Note: If set to <code>true</code> a <code>ui5-busy-indicator</code> component will be displayed into the related to the current <code>ui5-menu-item</code> sub-menu popover.
```
Comment (DavidMitreski5) [inline packages/main/src/MenuItem.ts:103]: If set to <code>true</code> a <code>ui5-busy-indicator</code> component will be displayed into the related one to the current <code>ui5-menu-item</code> sub-menu popover.

File: packages/main/src/MenuItem.ts [hunk 7]
```diff
@@ -96,6 +97,32 @@ class MenuItem extends UI5Element {
...
+	 * @type {boolean}
+	 * @defaultvalue false
+	 * @public
+	 * @since 1.13.0
+	 */
+	@property({ type: Boolean })
+	busy!: boolean;
+
+	/**
+	 * Defines the delay in milliseconds, after which the busy indicator will displayed inside the corresponding ui5-menu popover.
```
Comment (DavidMitreski5) [inline packages/main/src/MenuItem.ts:115]: Defines the delay in milliseconds, after which the busy indicator will be displayed inside the corresponding ui5-menu popover.

**Review by DavidMitreski5 — CHANGES_REQUESTED** [review]

File: packages/main/src/Menu.hbs [hunk 8]
```diff
@@ -46,7 +46,15 @@
 	<div
 		id="{{_id}}-menu-main"
 	>
-	{{#if _currentItems.length}}
+
+	{{#if busy}}
```
[the remark this answers is not in the document]
Comment (unazko) [reply packages/main/src/Menu.hbs]: The initial concept for this feature was that all menu items corresponding to a given menu would be loaded from a database. The described case is an additional scenario when more items would be loaded and displayed alongside the already existing ones. As we discussed this additional scenario is now also implemented.

File: packages/main/src/Menu.ts (hunk 5 above)
Comment (unazko) [reply packages/main/src/Menu.ts]: done

File: packages/main/src/MenuItem.ts (hunk 6 above)
Comment (unazko) [reply packages/main/src/MenuItem.ts]: done

File: packages/main/src/MenuItem.ts (hunk 7 above)
Comment (unazko) [reply packages/main/src/MenuItem.ts:115]: done

File: packages/main/src/MenuItem.ts [hunk 9]
```diff
@@ -96,6 +97,32 @@ class MenuItem extends UI5Element {
...
```
[the remark this answers is not in the document]
Comment (unazko) [reply packages/main/src/MenuItem.ts]: done

**Review by DavidMitreski5 — APPROVED** [review]

**Review by NHristov-sap — APPROVED** [review]

**Pull request merged**